v2/internal/controllers/to_azure_secrets_test.go

```diff
@@ -31,20 +32,32 @@ func Test_MissingUserSecret_ReturnsError(t *testing.T) {
 	// https://github.com/Azure/azure-service-operator/issues/1944
 	tc.CreateResourceAndWait(vnet)
 	tc.CreateResourcesAndWait(subnet, networkInterface)
-	secret := genruntime.SecretReference{
+	secretRef := genruntime.SecretReference{
 		Name: "thisdoesntexist",
 		Key:  "key",
 	}
-	vm := newVM(tc, rg, networkInterface, secret)
+	vm := newVM(tc, rg, networkInterface, secretRef)
 
 	tc.CreateResourceAndWaitForState(vm, metav1.ConditionFalse, conditions.ConditionSeverityWarning)
 	// We expect the ready condition to include details of the error
 	tc.Expect(vm.Status.Conditions[0].Reason).To(Equal("SecretNotFound"))
 	tc.Expect(vm.Status.Conditions[0].Message).To(
-		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secret.Name))
+		ContainSubstring("failed resolving secret references: %s/%s does not exist", tc.Namespace, secretRef.Name))
+
+	// Now create the secret
+	secret := &v1.Secret{
+		ObjectMeta: tc.MakeObjectMetaWithName(secretRef.Name),
+		StringData: map[string]string{
+			secretRef.Key: tc.Namer.GeneratePasswordOfLength(40),
+		},
+	}
+	tc.CreateResource(secret)
+
+	// Wait for the VM to make it to a steady state
+	tc.Eventually(vm).Should(tc.Match.BeProvisioned(0))
 
 	// Delete VM and resources.
-	tc.DeleteResourcesAndWait(vm, networkInterface, subnet, vnet, rg)
+	tc.DeleteResourcesAndWait(rg)
```

Comment:
Shouldn't this be deleting the virtual machine and leaving deletion of the resource group to our test framework?

Reply:
Fixed